nmdc_automation/import_automation/activity_mapper.py

```diff
@@ -52,140 +51,169 @@ def __init__(
         self.project_dir = project_directory
         self.url = self.import_data["Workflow Metadata"]["Source URL"]
         self.data_object_type = "nmdc:DataObject"
-        self.objects = {}
+        self.data_object_map = {}
         self.workflow_execution_ids = {}
-        self.workflows_by_type = {}
+        self.workflows_by_type = self.build_workflows_by_type()
+        self.runtime = runtime
 
-        self.runtime = NmdcRuntimeApi(site_config_file)
 
-        for wf in self.import_data["Workflows"]:
-            self.workflows_by_type[wf["Type"]] = wf
+    def load_yaml_file(self, yaml_file: Union[str, Path]) -> Dict:
+        """Utility function to load YAML file."""
+        with open(yaml_file, "r") as file:
+            return yaml.safe_load(file)
+
+    def build_workflows_by_type(self) -> Dict:
+        """Builds a dictionary of workflows by their type."""
+        return {wf["Type"]: wf for wf in self.import_data["Workflows"]}
 
-    def unique_object_mapper(self) -> None:
+    def map_sequencing_data(self) -> Tuple[nmdc.Database, Dict]:
         """
-        Map unique data objects from the file list based on unique matching import suffix.
-        The method relates each object to an workflow execution ID and updates the file with object action.
-        It updates the nmdc database with the DataObject and stores the information in the objects dictionary.
+        Map sequencing data to an NMDC data object and create an update to be applied to the has_output
+        list of the sequencing data generation.
         """
-
-        for data_object_dict in self.import_data["Data Objects"]["Unique"]:
-            if not filter_import_by_type(
-                self.import_data["Workflows"], data_object_dict["output_of"]
-            ):
-                continue
-            for file in self.file_list:
-                file = str(file)
-                if data_object_dict is None:
-                    continue
-                elif "import_suffix" not in data_object_dict:
-                    logging.warning("Missing suffix")
-                    continue
-
-                elif re.search(data_object_dict["import_suffix"], file):
-                    workflow_execution_id = self.get_workflow_execution_id(data_object_dict["output_of"])
-
+        sequencing_types = ["Metagenome Raw Reads", "Metatranscriptome Raw Reads"]
```

Comment:
"Metatranscriptome Raw Reads" is not currently a permissible value in FileTypeEnum.

Reply:
Removed